crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -630,6 +638,38 @@ pub fn evaluate_cmp(
     }
 }
 
+//Performs byte decomposition
+pub fn to_bytes(lhs: &InternalVar, bit_size: u32, evaluator: &mut Evaluator) -> Vec<Witness> {
+    assert!(bit_size < FieldElement::max_num_bits());
+    let mut bytes = Expression::default();
+    let mut two_pow = FieldElement::one();
+    let two = FieldElement::from(2_i128);
+    let mut result = Vec::new();
+    for _ in 0..(bit_size / 8) {
+        let byte_witness = evaluator.add_witness_to_cs();
+        result.push(byte_witness);
+        let byte_expr = from_witness(byte_witness);
+        bytes = add(&bytes, two_pow, &byte_expr);
+        for _ in 0..8 {
+            two_pow = two_pow.mul(two);
+        }
+        evaluator.gates.push(Gate::Arithmetic(subtract(
+            &byte_expr,
+            FieldElement::one(),
+            &byte_expr,
+        )));
+    }
+    evaluator.gates.push(Gate::Directive(Directive::ToBytes {
```

Comment:
What about bit_size which are not a multiple of 8 or bigger than the modulus?

Reply:
For right now basically the last byte will be truncated if the bit_size is not a multiple of 8. And I have this assert statement, `assert!(bit_size < FieldElement::max_num_bits());`, at the top of `to_bytes`. It currently still follows what we are doing in the `split` function where the max bit size is 253 so it cannot be bigger than the modulus. 

I was thinking that checking directly against the modulus rather than the bit size might be better scoped in a separate PR.

Reply:
Thank you for your answer, I did not saw the assert, sorry.
Regarding the truncation if not multiple of 8, I think this is not good as it might fail and users will not know why, so you should either handle this case of force users to provide a multiple of 8 (i.e a byte_size rather than a bit_size)

Reply:
Ok. I am leaning towards requiring users to supply a byte size rather than a bit size. However, I am also trying to think of how to best handle this case for `to_radix` when we have different multiples, not just multiples of 8.

Reply:
You could have the to_bytes in the stdlib, which requires a byte size, and this function uses to_radix internally.